pairing/bn256/scalar.go

```diff
@@ -116,8 +125,14 @@ func (s *scalar) SetVarTime(varTime bool) error {
 
 func (s *scalar) MarshalBinary() ([]byte, error) {
 	n := s.MarshalSize()
-	buf := s.x.Bytes()
-	return buf[:n], nil
+	buf := make([]byte, n)
+	bytes := s.x.Bytes()
+	if n < len(bytes) {
+		return nil, errors.New("bn256.Scalar: unexpected size")
+	}
+	m := n - len(bytes)
+	copy(buf[m:n], bytes)
```

Comment:
Question: is `s.x.Bytes()` little- or big-endian? Here you suppose it's big-endian, no?

Reply:
I was sticking to the Big.Int behavior which does big-endian, see https://golang.org/pkg/math/big/#Int.Bytes.